web/src/app/shared/store/classification-store/classification.state.ts

```diff
@@ -1,46 +1,187 @@
-import { SetSelectedClassificationType } from './classification.actions';
+import { Action, NgxsAfterBootstrap, State, StateContext } from '@ngxs/store';
+import { Observable, of } from 'rxjs';
+import { take, tap } from 'rxjs/operators';
+import { CategoriesResponse,
+  ClassificationCategoriesService } from '../../services/classification-categories/classification-categories.service';
+import { CreateClassification,
+  DeselectClassification,
+  GetClassifications,
+  RemoveSelectedClassification,
+  RestoreSelectedClassification,
+  SaveClassification,
+  SelectClassification,
+  SetActiveAction,
+  SetSelectedClassificationType,
+  UpdateClassification } from './classification.actions';
+import { ClassificationsService } from '../../services/classifications/classifications.service';
+import { ClassificationDefinition } from '../../models/classification-definition';
+import { ACTION } from '../../models/action';
 
 class InitializeStore {
   static readonly type = '[ClassificationState] Initializing state';
 }
 
 @State<ClassificationStateModel>({ name: 'classification' })
-export class ClassificationState {
-  constructor(private categoryService: ClassificationCategoriesService) {
+export class ClassificationState implements NgxsAfterBootstrap {
+  constructor(private categoryService: ClassificationCategoriesService,
+    private classificationsService: ClassificationsService) {
   }
 
   @Action(SetSelectedClassificationType)
-  setSelectedClassificationType(ctx: StateContext<ClassificationStateModel>, action: SetSelectedClassificationType): void {
+  setSelectedClassificationType(ctx: StateContext<ClassificationStateModel>, action: SetSelectedClassificationType): Observable<null> {
     const state: ClassificationStateModel = ctx.getState();
     if (state.classificationTypes[action.selectedType]) {
       ctx.patchState({ selectedClassificationType: action.selectedType });
     }
+    return of(null);
+  }
+
+  @Action(SelectClassification)
+  selectClassification(ctx: StateContext<ClassificationStateModel>, action: SelectClassification): Observable<any|null> {
+    if (typeof action.classificationId !== 'undefined') {
+      return this.classificationsService.getClassification(action.classificationId).pipe(take(1), tap(
+        selectedClassification => {
+          ctx.patchState({
+            selectedClassification,
+            action: null
+          });
+        }
+      ));
+    }
+    return of(null);
+  }
+
+  @Action(DeselectClassification)
+  deselectClassification(ctx: StateContext<ClassificationStateModel>): Observable<null> {
+    ctx.patchState({
+      selectedClassification: undefined,
+      action: null
+    });
+    return of(null);
   }
 
   @Action(InitializeStore)
   initializeStore(ctx: StateContext<ClassificationStateModel>): Observable<any> {
     return this.categoryService.getClassificationCategoriesByType().pipe(
-      tap(classificationTypes => {
+      take(1), tap(classificationTypes => {
```

Comment:
Why do we need this take(1) here? I thought ngxs will automatically subscribe and directly unsubscribe after the first value.